src/clients/HttpTunneler.cc

```diff
@@ -374,10 +374,12 @@ Http::Tunneler::sendSuccess()
 }
 
 void
-Http::Tunneler::countFailingConnection(const ErrorState * const error)
+Http::Tunneler::countFailingConnection()
 {
     assert(connection);
-    NoteOutgoingConnectionFailure(connection->getPeer(), error ? error->httpStatus : Http::scNone);
```

Comment:
This single call removal is the primary fix. Everything else is related cleanup.